palm/plugins/dbt/plugin_config.py

```diff
@@ -2,13 +2,31 @@
 
 from typing import Optional
 from pathlib import Path
+import semver
 from pydantic import BaseModel
 from palm.plugins.base_plugin_config import BasePluginConfig
+from palm.plugins.dbt.dbt_version_detection import get_dbt_version
 
 
 class dbtPluginConfigModel(BaseModel):
     dbt_artifacts_prod: Optional[str]
     dbt_artifacts_local: str
+    dbt_version: str
```

Comment:
We could do this:
```suggestion
    dbt_version: str = Field(default_factory=get_dbt_version)
```
to prevent missing config errors when upgrading, but would default to spinning up a docker container to get the version every time the plugin config is read... which isn't ideal. I am also open to changing `BasePluginConfig._read` to `BasePluginConfig.read` and making it an `@abstractmethod` so that it can be overridden, which would allow us to add something like:

```
if not plugin_config.model:
            _, plugin_config = self.update()
```

Or maybe we just need a better implementation of `validate`?

I'll think on this some over the weekend. Thoughts and suggestions are very much welcome.

Reply:
What validate method are you referring to?

Did your weekend thinking clear up anything for you?

I like the abstract method option you suggested, but I'm not confident about my understanding of any implications of doing that vs the other options.

Reply:
This is the validate method: https://github.com/palmetto/palm-cli/blob/develop/palm/plugins/base_plugin_config.py#L59

I just pushed up an update to this PR which I think it pretty neat... not perfect, but a good improvement.